src/Appwrite/Platform/Workers/Databases.php

```diff
@@ -1,76 +1,94 @@
 <?php
 
+namespace Appwrite\Platform\Workers;
+
 use Appwrite\Event\Event;
-use Appwrite\Extend\Exception;
 use Appwrite\Messaging\Adapter\Realtime;
-use Appwrite\Resque\Worker;
+use Exception;
 use Utopia\CLI\Console;
 use Utopia\Database\Database;
 use Utopia\Database\Document;
+use Utopia\Database\Exception\Authorization;
+use Utopia\Database\Exception\Conflict;
+use Utopia\Database\Exception\Structure;
 use Utopia\Database\Exception as DatabaseException;
+use Utopia\Platform\Action;
+use Utopia\Queue\Message;
 
-require_once __DIR__ . '/../init.php';
-
-Console::title('Database V1 Worker');
-Console::success(APP_NAME . ' database worker v1 has started' . "\n");
-
-class DatabaseV1 extends Worker
+class Databases extends Action
 {
-    public function init(): void
+    public static function getName(): string
     {
+        return 'databases';
     }
 
-    public function run(): void
+    /**
+     * @throws Exception
+     */
+    public function __construct()
     {
-        $type = $this->args['type'];
-        $project = new Document($this->args['project']);
-        $collection = new Document($this->args['collection'] ?? []);
-        $document = new Document($this->args['document'] ?? []);
-        $database = new Document($this->args['database'] ?? []);
+        $this
+            ->desc('Databases worker')
+            ->inject('message')
+            ->inject('dbForConsole')
+            ->inject('dbForProject')
```

Comment:
Are you sure you can directly use `dbForProject` here? 

What is the `projectId` when this resource is created? From what I remember in the code ,we dont know the project ID until we parse `$this->args['project']`

Reply:
dbForProject is with the project arg

Server::setResource('dbForProject', function (Cache $cache, Registry $register, Message $message, Database $dbForConsole) {
    $payload = $message->getPayload() ?? [];
    $project = new Document($payload['project'] ?? []);